cli/src/wallet.rs

```diff
@@ -1733,7 +1734,9 @@ fn build_balance_message(lamports: u64, use_lamports_unit: bool) -> String {
         format!("{:?} lamport{}", lamports, ess)
     } else {
         let sol = lamports_to_sol(lamports);
-        format!("{:.8} SOL", sol)
+        let sol_str = format!("{:.8}", sol);
```

Comment:
Does the precision parameter round here?  eg. `0.123_456_789` -> `0.123_456_79`

Reply:
Yes

Reply:
That tends to lead to confused users when they inevitably c/p the value and get an "Insufficient funds" error :slightly_smiling_face:

Reply:
Got a better suggestion? Tx fees come into play with c/p anyway...

Reply:
Working in lamports is always still an option.

Reply:
I'd usually do something along the lines of `int(N * 10e8) / 10e8`.  It looks like we're using `f64` in the lamports <-> SOL utils though, which already discards precision.  We'll want to move to a [Decimal floating point](https://en.wikipedia.org/wiki/Decimal_floating_point) crate eventually.  Numbers are hell in financial applications :pensive:.  A comment may be sufficient for now?